### DISCUSSION TITLE
dtOptions: DataTables.Settings - ignored

### DISCUSSION THREAD

**Discussion by giuseppemorale** [l-lin/angular-datatables#1629] [Q&A]
# :beetle: bug report

DataTables Options not working.
Tried to change Language (and others) with DataTables.Settings, it doesn't work

## :microscope: Minimal Reproduction

**StackBlitz/GitHub Link:**
- https://github.com/giuseppemorale/datatables-italian-options
- https://stackblitz.com/edit/datatables-italian-options (it doesn't read angular.json)

**Step-by-step Instructions:**

## :8ball: Expected behavior

Language Datatable Italian

## I've created a service to manage DataTables (i've also tried in component)

![service](https://user-images.githubusercontent.com/42799690/149599705-7c394fce-9334-43ed-a1cc-d90a48093894.png)
## ngOnInit in component:

![ngOnInit](https://user-images.githubusercontent.com/42799690/149599708-dd057f2c-e559-4605-bc08-1d3ae871a661.png)
## Table Result:

![dataTable](https://user-images.githubusercontent.com/42799690/149599709-84b6681d-2980-4910-8f6e-c682cb0cbe10.png)
## Html:

![component-html](https://user-images.githubusercontent.com/42799690/149599712-3c5e8171-36e7-4916-8539-222f4c63cb5f.png)

## :globe_with_meridians: Your Environment

- NodeJS version: v12.22.5
- Angular version: 13.1.0
- Angular CLI version: 13.1.0
- jQuery version: 3.6.0
- DataTables version: 1.11.3
- angular-datatables version: 13.0.1

## :memo: Additional context

CODE PASSED TO SETTINGS:
<pre>
<code>
public tableOptions: DataTables.Settings =  {
        pagingType: 'full_numbers',
        pageLength: 10,
        language: {
            emptyTable: "Nessun dato presente nella tabella",
            info: "Vista da _START_ a _END_ di _TOTAL_ elementi",
            infoEmpty: "Vista da 0 a 0 di 0 elementi",
            infoFiltered: "(filtrati da _MAX_ elementi totali)",
            infoPostFix: "",
            lengthMenu: "Visualizza _MENU_ elementi",
            loadingRecords: "Caricamento...",
            processing: "<i class='fas fa-spinner fa-spin text-dark' style='font-size:24px'></i>",
            search: "Cerca:",
            zeroRecords: "La ricerca non ha portato alcun risultato.",
            paginate: {
                first: "Inizio",
                previous: "Precedente",
                next: "Successivo",
                last: "Fine"
            },
            aria: {
                sortAscending: ": attiva per ordinare la colonna in ordine crescente",
                sortDescending: ": attiva per ordinare la colonna in ordine decrescente"
            }
      }
}
</code>
</pre>

**Comment by shanmukhateja** [comment]
Hi, please provide a reproducible StackBlitz link or GitHub repo to investigate this.

**Reply by giuseppemorale** [reply]
https://github.com/giuseppemorale/datatables-italian-options
https://stackblitz.com/edit/datatables-italian-options (it doesn't read angular.json scripts)

**Reply by shanmukhateja** [reply]
Hi @giuseppemorale this could be because DataTables Core plugin isn't seeing any data to actually render onto HTML. You're using Angular to display the tabular rows onto DOM which I suspect is why it doesn't work.

I would recommend using [ajax](https://datatables.net/reference/option/ajax) + [columns](https://datatables.net/reference/option/columns) to "inform" DataTables how to render the table all by itself (without Angular or user intervention like manually declaring `thead` or using ngFor in `tbody` on HTML)
